.travis.yml

```diff
@@ -1,12 +1,16 @@
 language: python
-sudo: false
-env:
-    - TOXENV=py26
-    - TOXENV=py27
-    - TOXENV=py33
+dist: xenial
+python:
+  - 2.7
+  - 3.5
+  - 3.6
+  - 3.7
 install:
-    - travis_retry pip install tox
+  - pip install zope.testrunner coverage coveralls
+  - pip install -e .[test]
 script:
-    - tox
+  - coverage run -m zope.testrunner --test-path=src -vc
+after_success:
+  - coveralls
 notifications:
-    email: false
+  email: false
```

Comment:
I never remember the right magic incantation.  I've seen several examples and I don't know which one's the best one to copy.

Reply:
`cache: pip` is sufficient these days for simple projects like this (that used to not be the case, though I don't remember exactly why off the top of my head). For bonus points, you can clean up pip's log to try to prevent spurious cache re-writes:
```yaml
 cache: pip
 before_cache:
   - rm -f $HOME/.cache/pip/log/debug.log
```